ports/esp32/modesp32.c

```diff
@@ -145,6 +148,32 @@ STATIC mp_obj_t esp32_hall_sensor(void) {
 }
 STATIC MP_DEFINE_CONST_FUN_OBJ_0(esp32_hall_sensor_obj, esp32_hall_sensor);
 
+STATIC mp_obj_t esp32_espidf_heap_info(void) {
+    #if 0
+    heap_caps_print_heap_info(MALLOC_CAP_8BIT);
+    return mp_obj_new_int(0);
+    #else
+    multi_heap_info_t info;
+    heap_t *heap;
+    mp_obj_t heap_list = mp_obj_new_list(0, 0);
+    SLIST_FOREACH(heap, &registered_heaps, next) {
+        if (heap_caps_match(heap, MALLOC_CAP_8BIT)) {
```

Comment:
Would it be useful to list all memory types?  eg would be good to know available IRAM for loading native code in to.  In that case probably need to add an extra entry to the return tuple, at the start, being the memory type.